From 779a9eea23d1a369c219f3b73cfbf1c2c4a09554 Mon Sep 17 00:00:00 2001
From: Tingjia-0v0 <tjcao980311@gmail.com>
Date: Tue, 4 Nov 2025 14:22:02 -0600
Subject: [PATCH] 6.12-rc3-cd9626e_fix

---
 include/linux/sched.h | 5 +++++
 kernel/freezer.c      | 9 +++++++--
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/include/linux/sched.h b/include/linux/sched.h
index 449dd64ed9ac..bb343136ddd0 100644
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -2133,6 +2133,11 @@ static inline void set_task_cpu(struct task_struct *p, unsigned int cpu)
 
 #endif /* CONFIG_SMP */
 
+static inline bool task_is_runnable(struct task_struct *p)
+{
+	return p->on_rq && !p->se.sched_delayed;
+}
+
 extern bool sched_task_on_rq(struct task_struct *p);
 extern unsigned long get_wchan(struct task_struct *p);
 extern struct task_struct *cpu_curr_snapshot(int cpu);
diff --git a/kernel/freezer.c b/kernel/freezer.c
index 44bbd7dbd2c8..6d1a1629ef04 100644
--- a/kernel/freezer.c
+++ b/kernel/freezer.c
@@ -108,8 +108,13 @@ static void fake_signal_wake_up(struct task_struct *p)
 static int __set_task_frozen(struct task_struct *p, void *arg)
 {
 	unsigned int state = READ_ONCE(p->__state);
-
-	if (p->on_rq)
+	
+	/*
+	 * Allow freezing the sched_delayed tasks; they will not execute until
+	 * ttwu() fixes them up, so it is safe to swap their state now, instead
+	 * of waiting for them to get fully dequeued.
+	 */
+	 if (task_is_runnable(p))
 		return 0;
 
 	if (p != current && task_curr(p))
-- 
2.43.0

