diff --git a/arch/x86/kernel/tsc.c b/arch/x86/kernel/tsc.c
index 87e749106dda..cf93faa36502 100644
--- a/arch/x86/kernel/tsc.c
+++ b/arch/x86/kernel/tsc.c
@@ -283,6 +283,13 @@ bool using_native_sched_clock(void) { return true; }
 
 notrace u64 sched_clock(void)
 {
+	if (smp_processor_id() != 0) {
+		if (sched_clock_mocked_warned) {
+			pr_info("sched_clock: cpu=%d, sched_clock_mocked_value=%llu\n",
+				smp_processor_id(), sched_clock_mocked_value);
+		}
+		return sched_clock_mocked_value += 1000000;
+	}
 	u64 now;
 	preempt_disable_notrace();
 	now = sched_clock_noinstr();
diff --git a/include/linux/sched/clock.h b/include/linux/sched/clock.h
index 196f0ca351a2..aa07254b922b 100644
--- a/include/linux/sched/clock.h
+++ b/include/linux/sched/clock.h
@@ -4,6 +4,11 @@
 
 #include <linux/smp.h>
 
+extern bool sched_clock_mocked_warned;
+extern u64 sched_clock_mocked_value;
+extern void sched_clock_mock_set_warn(bool warned);
+extern void sched_clock_mock_set_value(u64 value);
+
 /*
  * Do not use outside of architecture code which knows its limitations.
  *
diff --git a/kernel/sched/clock.c b/kernel/sched/clock.c
index f5e6dd6a6b3a..a28b501d2ed9 100644
--- a/kernel/sched/clock.c
+++ b/kernel/sched/clock.c
@@ -57,6 +57,21 @@
 #include <linux/sched/clock.h>
 #include "sched.h"
 
+bool sched_clock_mocked_warned = false;
+u64 sched_clock_mocked_value = 0;
+
+void sched_clock_mock_set_warn(bool warned)
+{
+	sched_clock_mocked_warned = warned;
+}
+EXPORT_SYMBOL(sched_clock_mock_set_warn);
+
+void sched_clock_mock_set_value(u64 value)
+{
+	sched_clock_mocked_value = value;
+}
+EXPORT_SYMBOL(sched_clock_mock_set_value);
+
 /*
  * Scheduler clock - returns current time in nanosec units.
  * This is default implementation.
